index.js

```diff
@@ -555,9 +557,95 @@ const plugin = fp(async function (app, opts) {
       operationName
     })
 
+    /* istanbul ignore next */
+    if (execution.initialResult) {
+      const acceptHeader = reply.request.raw.headers.accept
+
+      if (
+        !(
+          acceptHeader &&
+          new Negotiator({
+            headers: { accept: acceptHeader }
+          }).mediaType([
+            // mediaType() will return the first one that matches, so if the client
+            // doesn't include the deferSpec parameter it will match this one here,
+            // which isn't good enough.
+            MEDIA_TYPES.MULTIPART_MIXED_NO_DEFER_SPEC,
+            MEDIA_TYPES.MULTIPART_MIXED_EXPERIMENTAL
+          ]) === MEDIA_TYPES.MULTIPART_MIXED_EXPERIMENTAL
+        )
+      ) {
+        // The client ran an operation that would yield multiple parts, but didn't
+        // specify `accept: multipart/mixed`. We return an error.
+        throw new Error(
+          'Mercurius server received an operation that uses incremental delivery ' +
+          '(@defer or @stream), but the client does not accept multipart/mixed ' +
+          'HTTP responses. To enable incremental delivery support, add the HTTP ' +
+          "header 'Accept: multipart/mixed; deferSpec=20220824'.",
+          // Use 406 Not Accepted TODO
+          { extensions: { http: { status: 406 } } }
+        )
+      }
```

Comment:
If I read this error correctly, in case the request was not made with multipart accept it will fail.
Can we fall back to the previous behavior instead, i.e. accumulate all results.

Reply:
Not sure I understand this. What do you mean by `previous behavior` and `accumulate all results`?

This line, btw, isn't necessary and I removed it. Although, I still throw an error [here](https://github.com/mercurius-js/mercurius/pull/898/files#diff-3fad171d5e074854125d96b9140d539b46dc2eae91814bf3b7b17fd99b2e2d55R93).

As I understand, there is no way of not executing `@defer` if it's in the query, so we have the deferred result anyway. Are you asking about just sending it back anyway, and not throwing an error if the `accept` header is incorrect?

Reply:
You are on the spot. I completely misunderstood this implementation and throwing is the correct behavior.